frontend/src/components/table/table-component.tsx

```diff
@@ -8,8 +8,8 @@ import TableHeader from './table-header';
 
 interface TableComponentProps {
   headerNames: string[];
-  onRowClick: (row: any) => void;
-  data: RowDataType[];
+  onRowClick?: (row: unknown) => void;
```

Comment:
i assume it's a mix between string or boolean?

could also use then 
```suggestion
  onRowClick?: (row: string | boolean) => void;
```